src/typegoose.ts

```diff
@@ -77,9 +77,17 @@ export class Typegoose {
       sch.statics = sch.statics || {};
     }
 
+    const initMethods = methods.initMethods[name];
     const instanceMethods = methods.instanceMethods[name];
-    if (instanceMethods) {
+    if (instanceMethods && initMethods) {
```

Comment:
why make a duplicate check if it could just be if (initMethods) and if (instanceMethods) instead of if (both) else (instanceMethods) else (initMethods)?